cli/src/config.rs

```diff
@@ -38,12 +38,23 @@ pub enum ConfigError {
 pub enum ConfigSource {
     Default,
     Env,
-    // TODO: Track explicit file paths, especially for when user config is a dir.
     User,
     Repo,
     CommandArg,
 }
 
+impl fmt::Display for ConfigSource {
+    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
```

Comment:
I would use a custom name for this function (instead of implementing a trait), since we may not always want to display a config source in the same way (for example, we could represent each source type by a single letter). But I'll leave it to people more experienced in the jj project than I to decide.

Reply:
Hmm, I see the benefit keeping our options open in regards to formatting.  Searching around in `cli/src` I generally see `fmt::Display` being used, e.g. [RemoteBranchName](https://github.com/martinvonz/jj/blob/dce99cf1f83bfc54a4130c4fe2b1c1d0a81a980c/cli/src/cli_util.rs#L2460), though there are exceptions like [CommitOrChangeId](https://github.com/martinvonz/jj/blob/dce99cf1f83bfc54a4130c4fe2b1c1d0a81a980c/cli/src/commit_templater.rs#L564-L580).

Let's see what others say.

Reply:
Other than `Error` types, I think `fmt::Display` is used where there's  one reasonable string representation.

For `ConfigSource` enum, I'm okay with either way. I personally would avoid implementing `fmt::Display` because the output (like `"command line"`) seems more verbose than I would expect from `fmt::Display` of enums, though.

Reply:
I've added a `description` method in lieu of the `fmt::Display` implementation.